docs/aws/audit/dynamodbmonitoring/rules/ddb_customer_kms_key.mdx

```diff
@@ -23,6 +23,229 @@ HIPAA, NIST, SOC2, GDPR, ISO27001
+To ensure that your AWS DynamoDB tables use KMS Customer Master Keys (CMKs) for encryption, you can follow these steps using the AWS CLI:
+
+1. **Create a KMS Key**:
+   First, create a KMS key that will be used to encrypt your DynamoDB table.
+   ```sh
+   aws kms create-key --description "DynamoDB encryption key"
+   ```
+
+2. **Get the KMS Key ID**:
+   Retrieve the Key ID of the newly created KMS key.
+   ```sh
+   aws kms list-keys
+   ```
+
+3. **Enable Encryption on a New DynamoDB Table**:
+   When creating a new DynamoDB table, specify the KMS key for encryption.
+   ```sh
+   aws dynamodb create-table \
+       --table-name MyTable \
+       --attribute-definitions AttributeName=Id,AttributeType=S \
+       --key-schema AttributeName=Id,KeyType=HASH \
+       --provisioned-throughput ReadCapacityUnits=5,WriteCapacityUnits=5 \
+       --sse-specification Enabled=true,SSEType=KMS,KMSMasterKeyId=<Your-KMS-Key-ID>
+   ```
+
+4. **Update an Existing DynamoDB Table to Use KMS Encryption**:
+   If you have an existing table, you can update it to use the KMS key for encryption.
+   ```sh
+   aws dynamodb update-table \
+       --table-name MyExistingTable \
+       --sse-specification Enabled=true,SSEType=KMS,KMSMasterKeyId=<Your-KMS-Key-ID>
+   ```
+
+Replace `<Your-KMS-Key-ID>` with the actual Key ID obtained from the `list-keys` command. These steps will ensure that your DynamoDB tables are encrypted using KMS CMKs.
+</Accordion>
+
+<Accordion title='Using Python'>
+To ensure that AWS DynamoDB tables use KMS Customer Master Keys (CMKs) for encryption, you can use the AWS SDK for Python (Boto3). Below are the steps to prevent this misconfiguration:
+
+1. **Install Boto3**:
+   Ensure you have Boto3 installed in your Python environment. If not, you can install it using pip:
+   ```bash
+   pip install boto3
+   ```
+
+2. **Set Up AWS Credentials**:
+   Make sure your AWS credentials are configured. You can set them up using the AWS CLI or by setting environment variables.
+
+3. **Create a Python Script**:
+   Write a Python script to create a DynamoDB table with KMS CMK encryption enabled.
+
+4. **Run the Script**:
+   Execute the script to create the DynamoDB table with the specified encryption settings.
+
+Here is a sample Python script to create a DynamoDB table with KMS CMK encryption:
+
+```python
+import boto3
+
+# Initialize a session using Amazon DynamoDB
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
+    region_name='YOUR_REGION'
+)
+
+# Initialize DynamoDB resource
+dynamodb = session.resource('dynamodb')
+
+# Define the table name and key schema
+table_name = 'YourTableName'
+key_schema = [
+    {
+        'AttributeName': 'PrimaryKey',
+        'KeyType': 'HASH'  # Partition key
+    }
+]
+
+# Define the attribute definitions
+attribute_definitions = [
+    {
+        'AttributeName': 'PrimaryKey',
+        'AttributeType': 'S'  # String type
+    }
+]
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>